sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -18,27 +23,147 @@ namespace Azure.Messaging.EventHubs
+        ///   as an aid to consumers who wish to deserialize the binary data.
         /// </remarks>
         ///
         /// <seealso cref="BinaryData" />
         /// <seealso cref="EventData.Properties" />
         ///
-        public BinaryData EventBody { get; }
+        public BinaryData EventBody => _amqpMessage.GetEventBody();
+
+        /// <summary>
+        ///   A MIME type describing the data contained in the <see cref="EventBody" />,
+        ///   intended to allow consumers to make informed decisions for inspecting and
+        ///   processing the event.
+        /// </summary>
+        ///
+        /// <value>
+        ///   The MIME type of the <see cref="EventBody" /> content; when unknown, it is
+        ///   recommended that this value should not be set.  When the body is known to be
+        ///   truly opaque binary data, it is recommended that "application/octet-stream" be
+        ///   used.
+        /// </value>
+        ///
+        /// <remarks>
+        ///   The <see cref="ContentType" /> is intended to allow coordination between event
+        ///   producers and consumers.  It has no meaning to the Event Hubs service.
+        /// </remarks>
+        ///
+        /// <seealso href="https://datatracker.ietf.org/doc/html/rfc2046">RFC2046 (MIME Types)</seealso>
+        ///
+        public string ContentType
+        {
+            get
+            {
+                if (_amqpMessage.HasSection(AmqpMessageSection.Properties))
+                {
+                    return _amqpMessage.Properties.ContentType?.ToString();
+                }
+
+                return null;
+            }
+
+            set
+            {
+                var populated = (!string.IsNullOrEmpty(value));
+
+                if ((populated) || (_amqpMessage.HasSection(AmqpMessageSection.Properties)))
+                {
+                    _amqpMessage.Properties.ContentType = populated
+                        ? value
+                        : null;
+                }
+            }
+        }
+
+        /// <summary>
+        ///   An application-defined value that uniquely identifies the event.  The identifier is
+        ///   a free-form value and can reflect a GUID or an identifier derived from the application
+        ///   context.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   The <see cref="MessageId" /> is intended to allow coordination between event
+        ///   producers and consumers.  It has no meaning to the Event Hubs service, and does
+        ///   not influence how Event Hubs identifies the event.
+        /// </remarks>
+        ///
+
+        public string MessageId
+        {
+            get
+            {
+                if (_amqpMessage.HasSection(AmqpMessageSection.Properties))
+                {
+                    return _amqpMessage.Properties.MessageId?.ToString();
+                }
+
+                return null;
```

Comment:
Could this be simplified to?:

```suggestion
                    return _amqpMessage.Properties.MessageId?.ToString();
```

Reply:
The message section are lazily allocated when the accessor is called.   If we were to go with the pattern that you're suggesting, it would work functionally, but would force an unwanted allocation in the event that the properties section was not created/populated.